lib/tree.rb

```diff
@@ -16,47 +16,126 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  def add_helper(node, key, value)
+    if node.nil?
+      node = TreeNode.new(key, value)
+    elsif key < node.key
+      node.left = add_helper(node.left, key, value)
+    elsif key > node.key
+      node.right = add_helper(node.right, key, value)
+    end
+    return node
+  end
+
+  # Time Complexity: O(1)
+  # Space Complexity: O(1)
   def add(key, value)
```

Comment:
Since `add_helper` has a time/space complexity of O(log n), this method does as well.